### PR TITLE
Feature/pentx upgrade

### PR DESCRIPTION
**Pull request by KrzysztofLabus-RDX**

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -2,34 +2,34 @@
...
+    <PackageVersion Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.3" />
     <PackageVersion Include="Polly" Version="7.2.3" />
-    <PackageVersion Include="prometheus-net" Version="7.0.0" />
-    <PackageVersion Include="prometheus-net.AspNetCore" Version="7.0.0" />
-    <PackageVersion Include="prometheus-net.AspNetCore.HealthChecks" Version="7.0.0" />
-    <PackageVersion Include="RadixDlt.RadixEngineToolkit" Version="0.8.0-build-17" />
+    <PackageVersion Include="prometheus-net" Version="8.0.0" />
+    <PackageVersion Include="prometheus-net.AspNetCore" Version="8.0.0" />
+    <PackageVersion Include="prometheus-net.AspNetCore.HealthChecks" Version="8.0.0" />
+    <PackageVersion Include="RadixDlt.RadixEngineToolkit" Version="0.8.1" />
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply Directory.Packages.props]: Updated

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/PendingTransactionResubmissionService.cs [hunk 2]
```diff
@@ -229,7 +224,7 @@ DateTime submittedAt
 
                 transaction.MarkAsRejected(
                     true,
-                    $"The transaction keeps dropping out of the mempool, so we're not resubmitting it. Initial failure reason: {transaction.FailureReason}",
+                    $"The transaction keeps dropping out of the mempool, so we're not resubmitting it. Initial failure reason: {transaction.LastFailureReason}",
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/PendingTransactionResubmissionService.cs:227]: We should most likely keep track of all statuses/failures, not just initial/last one - but that's something for this much bigger rework of PenTX, I don't want to blow this PR too much.

File: src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/CoreApiErrorWrapper.cs [hunk 3]
```diff
@@ -79,6 +79,27 @@ public static string GetDetailedExceptionName(this WrappedCoreApiException wrapp
         return $"{nameof(WrappedCoreApiException)}<{wrappedCoreApiException.Error.GetType().Name}>";
     }
 
+    public static async Task<ResponseOrError<TResponse, TError>> ResultOrError<TResponse, TError>(Func<Task<TResponse>> requestAction)
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/CoreApiErrorWrapper.cs:82]: Yeah, I'm aware of this library and few others but I'm not going to include it right now as a) we don't need universal tool yet, b) I'd like to reduce lib surface, c) if we'll find out that we're repeating the patter we should reevaluate this, what do you think?

File: src/RadixDlt.NetworkGateway.GatewayApi/Services/SubmissionService.cs [hunk 4]
```diff
@@ -205,60 +204,57 @@ private async Task<ToolkitModel.Transaction.NotarizedTransaction> HandlePreSubmi
                 finalTokenSource.Token
             );
 
-            if (result.Duplicate)
+            if (result.Succeeded)
             {
-                await _observers.ForEachAsync(x => x.SubmissionDuplicate(request, result));
+                var response = result.Success;
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Services/SubmissionService.cs]: Hmmm... I'm not convinced this is truly needed but then again it doesn't really hurts us to rename them, changed.

File: src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/CoreApiErrorWrapper.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.Abstractions/CoreCommunications/CoreApiErrorWrapper.cs:82]: that's exactly what I had in mind, just wanted to mention that lib if it gets bigger.

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**